tests/yam/validate/validate_minimal_system.pm

```diff
@@ -0,0 +1,41 @@
+# SUSE's openQA tests
+#
+# Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Validate the base product from /etc/products.d/baseproduct.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "consoletest";
+use testapi;
+use serial_terminal 'select_serial_terminal';
+
+sub run {
+    select_serial_terminal();
+
+    my @installed_packages = split /\n/, script_output('rpm -qa');
+    my %is_installed = map { $_ => 1 } @installed_pkgs;
+
+    record_info("Package number:", scalar @installed_pkgs);
+    record_info('Installed packages:', join("\n", @installed_packages));
+
+    my @all_recommends;
+    for my $pattern (split /\n/, script_output('rpm -qa patterns-*')) {
+        my @recommends = split /\n/, script_output("rpm -q --recommends $pattern");
```

Comment:
why not get the weak dependencies of all the packages instead of picking only the patterns (packages)?

Reply:
Additionally you can get rid off the intermediate variable `@recommends` having something like:
```
my $get_output_lines = sub {
    my ($command) = @_;
    return split /\n/, script_output($command);
};
```

Reply:
Changed.